services/schema.service.ts

```diff
@@ -142,7 +142,7 @@ export class SchemaService {
 
     private static isRequiredField(field: Field): boolean {
         if (typeof field === 'object') {
-            return 'constraints' in field && 'required' in field.constraints;
+            return field.required;
```

Comment:
I'd put a !! in front of field.required to make sure it returns a boolean (in case required is undefined).
```suggestion
            return !!field.required;
```

Reply:
`require` should not be undefined if `field` is really a frictionless tableschema field:
https://github.com/frictionlessdata/tableschema-js#field

But it doesn't hurt to add guards.

Reply:
done